xds/server.go

```diff
@@ -128,14 +128,32 @@ func NewGRPCServer(opts ...grpc.ServerOption) *GRPCServer {
 
 // handleServerOptions iterates through the list of server options passed in by
 // the user, and handles the xDS server specific options.
-func handleServerOptions(opts []grpc.ServerOption) *serverOptions {
+func (s *GRPCServer) handleServerOptions(opts []grpc.ServerOption) {
 	so := &serverOptions{}
 	for _, opt := range opts {
 		if o, ok := opt.(*serverOption); ok {
 			o.apply(so)
 		}
 	}
-	return so
+
+	// If the application did not register a mode change callback, the XdsServer
+	// registers its own which simply logs any errors and each serving resumption
+	// after an error, all at a default-visible log level, as per A36. The
+	// default-visible log level for us is ERROR.
+	//
+	// Note that this means that `s.opts.modeCallback` will never be nil and can
+	// safely be invoked directly from `handleServiceModeChanges`.
+	if so.modeCallback == nil {
```

Comment:
Should we have a `defaultServerOptions` that has this set in it, such that if applying the options doesn't override it, we get the defaults?  (e.g. how we handle DialOptions in the grpc package)

Reply:
Done.